From b2682485407967fd9b7101b28093e39b4fadcf3c Mon Sep 17 00:00:00 2001
From: "env." <162432464+belowzeroiq@users.noreply.github.com>
Date: Sat, 9 Aug 2025 07:56:42 +0700
Subject: [PATCH] scripts: skip android version and kmi strings

---
 scripts/setlocalversion | 29 ++++-------------------------
 1 file changed, 4 insertions(+), 25 deletions(-)

diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index 9887095a237b..fafc9b88536a 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -11,15 +11,13 @@
 #
 
 usage() {
-	echo "Usage: $0 [--save-scmversion] [--save-tag] [srctree] [branch] [kmi-generation]" >&2
+	echo "Usage: $0 [--save-scmversion] [--save-tag] [srctree]" >&2
 	exit 1
 }
 
 scm_only=false
 save_tag=false
 srctree=.
-android_release=
-kmi_generation=
 if test "$1" = "--save-scmversion"; then
 	scm_only=true
 	shift
@@ -32,22 +30,6 @@ if test $# -gt 0; then
 	srctree=$1
 	shift
 fi
-if test $# -gt 0; then
-	# Extract the Android release version. If there is no match, then return 255
-	# and clear the var $android_release
-	android_release=`echo "$1" | sed -e '/android[0-9]\{2,\}/!{q255}; \
-		s/^\(android[0-9]\{2,\}\)-.*/\1/'`
-	if test $? -ne 0; then
-		android_release=
-	fi
-	shift
-
-	if test $# -gt 0; then
-		kmi_generation=$1
-		[ $(expr $kmi_generation : '^[0-9]\+$') -eq 0 ] && usage
-		shift
-	fi
-fi
 if test $# -gt 0 -o ! -d "$srctree"; then
 	usage
 fi
@@ -69,11 +51,8 @@ scm_version()
 	# Check for git and a git repo.
 	if head=$(git rev-parse --verify HEAD 2>/dev/null); then
 
-		if [ -n "$android_release" ] && [ -n "$kmi_generation" ]; then
-			printf '%s' "-$android_release-$kmi_generation"
-		elif [ -n "$android_release" ]; then
-			printf '%s' "-$android_release"
-		fi
+		# Android release and KMI generation logic removed
+
 		# If we are at a tagged commit (like "v2.6.30-rc6"), by default
 		# we ignore it and manually define VERSION in the top level
 		# Makefile. Alternatively, set --save-tag to keep the tag.
@@ -176,4 +155,4 @@ if test -n "${BUILD_NUMBER}"; then
 	res="$res-ab${BUILD_NUMBER}"
 fi
 
-echo "$res"
+echo "$res"
\ No newline at end of file
-- 
2.50.1

