From aed9ef2428368e7ee556fcee808c6fad3bcfba17 Mon Sep 17 00:00:00 2001
From: "env." <162432464+belowzeroiq@users.noreply.github.com>
Date: Sat, 14 Feb 2026 18:33:23 +0700
Subject: [PATCH] Do not ship full autolocalversion

---
 scripts/setlocalversion | 25 ++++++++++++++++++++++++-
 1 file changed, 24 insertions(+), 1 deletion(-)

diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index 180764946a616..e6ad5d9ee7ede 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -11,13 +11,15 @@
 #
 
 usage() {
-	echo "Usage: $0 [--save-scmversion] [--save-tag] [srctree]" >&2
+	echo "Usage: $0 [--save-scmversion] [--save-tag] [srctree] [branch] [kmi-generation]" >&2
 	exit 1
 }
 
 scm_only=false
 save_tag=false
 srctree=.
+android_release=
+kmi_generation=
 if test "$1" = "--save-scmversion"; then
 	scm_only=true
 	shift
@@ -30,6 +32,22 @@ if test $# -gt 0; then
 	srctree=$1
 	shift
 fi
+if test $# -gt 0; then
+	# Extract the Android release version. If there is no match, then return 255
+	# and clear the var $android_release
+	android_release=`echo "$1" | sed -e '/android[0-9]\{2,\}/!{q255}; \
+		s/^\(android[0-9]\{2,\}\)-.*/\1/'`
+	if test $? -ne 0; then
+		android_release=
+	fi
+	shift
+
+	if test $# -gt 0; then
+		kmi_generation=$1
+		[ $(expr $kmi_generation : '^[0-9]\+$') -eq 0 ] && usage
+		shift
+	fi
+fi
 if test $# -gt 0 -o ! -d "$srctree"; then
 	usage
 fi
@@ -51,6 +69,11 @@ scm_version()
 	# Check for git and a git repo.
 	if head=$(git rev-parse --verify HEAD 2>/dev/null); then
 
+		# if [ -n "$android_release" ] && [ -n "$kmi_generation" ]; then
+		# 	printf '%s' "-$android_release-$kmi_generation"
+		# elif [ -n "$android_release" ]; then
+		#	printf '%s' "-$android_release"
+		# fi
 		# If we are at a tagged commit (like "v2.6.30-rc6"), by default
 		# we ignore it and manually define VERSION in the top level
 		# Makefile. Alternatively, set --save-tag to keep the tag.
-- 
2.52.0

